cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -1193,23 +1195,52 @@ public void run() {
     }
 
     /**
-     * Sets up resources usage for the task represented by this task type.
+     * Sets up the configuration for the task represented by this type.
      *
      * @param conf configuration to modify
+     * @param cConf the cdap configuration
+     * @param runtimeArgs the runtime arguments applicable for this task type
      * @param resources resources information or {@code null} if nothing to set
      */
-    public void setResources(Configuration conf, @Nullable Resources resources) {
+    public void configure(Configuration conf, CConfiguration cConf,
+                          Map<String, String> runtimeArgs, @Nullable Resources resources) {
       if (resources == null) {
```

Comment:
This means you can not configure using runtime arguments if the resources are null? When are they non-null? Only if they were configured at app deploy time?

Reply:
It is null if and only if resources is not set from the spec (deploy time) and not set via runtime arguments (`system.resources.memory`). On a hindsight, however, I think it should get default to what's set in the mapred conf. Will make the adjust to that.